cvat-ui/src/components/annotation-page/standard-workspace/controls-side-bar/tools-control.tsx

```diff
@@ -1022,32 +1022,89 @@ export class ToolsControlComponent extends React.PureComponent<Props, State> {
             });
         });
 
+        function checkAttributesCompatibility(
+            functionAttribute: ModelAttribute | undefined, // model
+            dbAttribute: any, // job
+            value: string,
+        ): boolean {
+            if (functionAttribute === undefined || dbAttribute === undefined) {
+                return false;
+            }
+
+            if (functionAttribute.input_type === 'checkbox') {
+                return ['true', 'false'].includes(value);
+            }
+
+            if (functionAttribute.input_type === dbAttribute.inputType) {
+                if (functionAttribute.input_type === 'number') {
+                    return !Number.isNaN(value);
+                }
+
+                return ['select', 'radio', 'text'].includes(functionAttribute.input_type);
+            }
+
+            switch (functionAttribute.input_type) {
```

Comment:
Let's move `functionAttribute.input_type === 'checkbox'` case from if on line 1034 here, some duplications are possible but I think its normal for the sake of readability